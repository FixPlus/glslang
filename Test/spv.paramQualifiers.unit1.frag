#version 460 core

float foo(in float f, out int i, inout vec3 v)
{
    v.x = f;
    i = int(v.y);
    return v.z;
}
