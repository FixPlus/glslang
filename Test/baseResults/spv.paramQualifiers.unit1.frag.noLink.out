spv.paramQualifiers.unit1.frag
Shader version: 460
gl_FragCoord origin is upper left
0:? Sequence
0:3  Function Definition: foo(f1;i1;vf3; ( global highp float)
0:3    Function Parameters: 
0:3      'f' ( in highp float)
0:3      'i' ( out highp int)
0:3      'v' ( inout highp 3-component vector of float)
0:5    Sequence
0:5      move second child to first child ( temp highp float)
0:5        direct index ( temp highp float)
0:5          'v' ( inout highp 3-component vector of float)
0:5          Constant:
0:5            0 (const int)
0:5        'f' ( in highp float)
0:6      move second child to first child ( temp highp int)
0:6        'i' ( out highp int)
0:6        Convert float to int ( temp highp int)
0:6          direct index ( temp highp float)
0:6            'v' ( inout highp 3-component vector of float)
0:6            Constant:
0:6              1 (const int)
0:7      Branch: Return with expression
0:7        direct index ( temp highp float)
0:7          'v' ( inout highp 3-component vector of float)
0:7          Constant:
0:7            2 (const int)
0:?   Linker Objects
0:?     Linker Function: foo(f1;i1;vf3; ( global highp float)
0:?       Function Parameters: 
0:?         '' ( in highp float)
0:?         '' ( out highp int)
0:?         '' ( inout highp 3-component vector of float)

Validation failed
// Module Version 10000
// Generated by (magic number): 80008
// Id's are bound by 28

                              Capability Shader
                              Capability Linkage
               1:             ExtInstImport  "GLSL.std.450"
                              MemoryModel Logical GLSL450
                              Source GLSL 460
                              Name 12  "foo(f1;i1;vf3;"
                              Name 9  "f"
                              Name 10  "i"
                              Name 11  "v"
                              Decorate 12(foo(f1;i1;vf3;) Linkage Attributes 678391654 1765486950 1719024433 15155 0
               2:             TypeFloat 32
               3:             TypePointer Function 2(float)
               4:             TypeInt 32 1
               5:             TypePointer Function 4(int)
               6:             TypeVector 2(float) 3
               7:             TypePointer Function 6(fvec3)
               8:             TypeFunction 2(float) 3(ptr) 5(ptr) 7(ptr)
              15:             TypeInt 32 0
              16:     15(int) Constant 0
              18:     15(int) Constant 1
              22:     15(int) Constant 2
              26:             TypeVoid
12(foo(f1;i1;vf3;):    2(float) Function None 8
            9(f):      3(ptr) FunctionParameter
           10(i):      5(ptr) FunctionParameter
           11(v):      7(ptr) FunctionParameter
              13:             Label
              14:    2(float) Load 9(f)
              17:      3(ptr) AccessChain 11(v) 16
                              Store 17 14
              19:      3(ptr) AccessChain 11(v) 18
              20:    2(float) Load 19
              21:      4(int) ConvertFToS 20
                              Store 10(i) 21
              23:      3(ptr) AccessChain 11(v) 22
              24:    2(float) Load 23
                              ReturnValue 24
                              FunctionEnd
